tests/utils/test_restart.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+"""Unit tests for the :py:mod:`~aiida_quantumespresso.utils.restart` module."""
+from __future__ import absolute_import
+
+import pytest
+
+from aiida.engine import ProcessBuilder
+from aiida_quantumespresso.utils import restart
+
+
+@pytest.fixture
+def generate_inputs():
+    """Return a dictionary of inputs for a `CalcJobNode` fixture to be created."""
+    from aiida import orm
+
+    inputs = {
+        'parameters': orm.Dict(dict={}),
+        'settings': orm.Dict(dict={})
+    }
+
+    return inputs
+
+
+class TestGetRestartBuilder(object):
+    """Tests for :py:mod:`~aiida_quantumespresso.utils.restart`."""
+
+    def test_restart(self, fixture_database, fixture_computer_localhost, generate_calc_job_node, generate_inputs):
```

Comment:
This function tests DosCalculation, but the function name and the docstring say something else.

Reply:
This is testing generic parts of the function, i.e. the type check. I just picked `DosCalculation` as one of the arbitrary calculation classes that are not supported

Reply:
Ah, I see. Then I would say that it specifically tests the failure case of an unsupported class, rather than "the generics". But ok, that's a minor complaint.

Reply:
Well the idea is that in the future more generics (behavior independent of process class of `node` argument) should be tested in this function. It is just that for the time being it is just testing the type check

Reply:
Ok.